### PR TITLE
Prototype of `ChildrenAs` processor

### PR DESCRIPTION
**Pull request by leekelleher**
Includes abstract processor `PublishedContentFilter`, which can be used to pass in a Where clause to filter out from a collection of nodes.

### CODE REVIEW

File: src/Our.Umbraco.Ditto.Contrib/Collections/PublishedContentByDocumentTypeAlias.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+﻿using Umbraco.Core;
+
+namespace Our.Umbraco.Ditto
+{
+    public class PublishedContentByDocumentTypeAlias : PublishedContentFilter
```
Comment (mattbrailsford) [inline src/Our.Umbraco.Ditto.Contrib/Collections/PublishedContentByDocumentTypeAlias.cs]: Seems a bit wordy, maybe something like DocTypeFilter would do?

File: src/Our.Umbraco.Ditto.Contrib/Collections/ChildrenAs.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
+﻿using System.Collections.Generic;
+
+namespace Our.Umbraco.Ditto
+{
+    public class ChildrenAs : DittoMultiProcessorAttribute
+    {
+        public ChildrenAs(string documentTypeAlias = null)
```
Comment (mattbrailsford) [inline src/Our.Umbraco.Ditto.Contrib/Collections/ChildrenAs.cs]: Wonder if we should have a constructorless version which just does [UmbracoProperty("Children")] as a convenience thing?

Comment (mattbrailsford) [inline src/Our.Umbraco.Ditto.Contrib/Collections/ChildrenAs.cs]: Maybe we should have a constructor that accepts an array of doc types? Not sure if this should then also tie into the DittoDocTypeFactory?

File: src/Our.Umbraco.Ditto.Contrib/Collections/PublishedContentFilter.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Umbraco.Core.Models;
+using Umbraco.Web;
+
+namespace Our.Umbraco.Ditto
+{
+    public abstract class PublishedContentFilter : DittoProcessorAttribute
```
Comment (mattbrailsford) [inline src/Our.Umbraco.Ditto.Contrib/Collections/PublishedContentFilter.cs]: Can't help but wonder if we could make a generic Filter attribute that can just filter any collection?

Comment (mattbrailsford) [reply src/Our.Umbraco.Ditto.Contrib/Collections/PublishedContentFilter.cs]: Might not be so easy though without generics support, but worth looking at.

**Review by mattbrailsford — COMMENTED** [review]
In all, I'm not against this, and can see the ChildrenAs attribute being a nice succinct solution if people don't want to write their own attribute for what I guess would be a common situation.

**Comment by leekelleher** [comment]
Merged in, we can refine in due course.
